package identifier;

import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;
import org.junit.Assert;

public class IdentifierTest_Functional
{
  protected Identifier id;

  @Before
  public void setUp() {
    id = new Identifier();
  }

  @Test 
  public void validateInitialSymbol1() {
    boolean result = id.validateIdentifier("Abcd5");
    Assert.assertEquals(true, result);
  }

  @Test
  public void validateInitialSymbol2() {
    boolean result = id.validateIdentifier("&123");
    Assert.assertFalse(result);
  }

  
  @Test
  public void validateMinLength1() {
	    boolean result = id.validateIdentifier("");
	    Assert.assertFalse(result);
  }

  @Test
  public void validateMinLength2() {
	    boolean result = id.validateIdentifier("a");
	    Assert.assertTrue(result);
  }


  @Test
  public void validateMaxLength1() {
    boolean result = id.validateIdentifier("123456");
    Assert.assertEquals(true, result);
  }

  @Test
  public void validateMaxLength2() {
    boolean result = id.validateIdentifier("1234567");
    Assert.assertEquals(true, result);
  }



  @Test
  public void validate4() {
    boolean result = id.validateIdentifier("Inv@lido");
    Assert.assertFalse(result);
  }


  @Test(expected=IndexOutOfBoundsException.class)
  public void stringException() {
    String str = new String("JUnit Example");
    str.substring(30);
  }

  @Test(timeout=2000)
  public void looping() {
    boolean result = id.validateIdentifier("Abcd5");
    Assert.assertEquals(true, result);
  }

  @Ignore("Out of the program scope")
  @Test(expected=IndexOutOfBoundsException.class)
  public void stringException2() {
    String str = new String("JUnit Example");
    str.substring(30);
  }
}
